source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -1066,10 +1065,12 @@ int pdo_sqlsrv_stmt_get_col_meta( _Inout_ pdo_stmt_t *stmt, _In_ zend_long colno
         // initialize the array to nothing, as PDO requires us to create it
         core::sqlsrv_array_init( *driver_stmt, return_value TSRMLS_CC );
 
-        sqlsrv_malloc_auto_ptr<field_meta_data> core_meta_data;
-
-        core_meta_data = core_sqlsrv_field_metadata( driver_stmt, (SQLSMALLINT) colno TSRMLS_CC );
+        field_meta_data* core_meta_data;
```

Comment:
Why are we not using the smart pointer here?

Reply:
Because if we use smart pointer the entire data will be transferred. See sqlsrv_malloc_auto_ptr definition